src/librustdoc/formats/renderer.rs

```diff
@@ -84,13 +88,22 @@ crate fn run_format<'tcx, T: FormatRenderer<'tcx>>(
             let _timer = prof.generic_activity_with_arg("render_mod_item", name.as_str());
 
             cx.mod_item_in(&item, &name)?;
-            let module = match *item.kind {
-                clean::StrippedItem(box clean::ModuleItem(m)) | clean::ModuleItem(m) => m,
+
+            if let Some(module) = match *item.kind {
+                clean::ModuleItem(m) => Some(m),
+                clean::StrippedItem(box clean::ModuleItem(m)) => {
+                    if T::RUN_ON_STRIPED {
+                        Some(m)
+                    } else {
+                        None
+                    }
+                }
```

Comment:
nit: I find this pretty hard to read, I would assign the option to a local variable instead of putting a `match` in the if condition.